backend/src/main/java/planit/people/preparation/Google_Connector/Google_Helper.java

```diff
@@ -81,9 +81,59 @@ public CalendarResponse createEvent(DTO_NewEventDetail newEventDetail) throws IO
 
         event = google_connector.calendar_service().events().insert(calendarId, event).execute();
         System.out.printf("Event created: %s\n", event.getHtmlLink());
-        return new CalendarResponse(startDateTime, endDateTime);
+        return new CalendarResponse(startDate, endDateTime);
+    }
+
+    public DateTime getStartDate(Date start_date, Date end_date, Long duration) throws IOException
+    {
+        org.joda.time.DateTime sD = new org.joda.time.DateTime(start_date);
+        org.joda.time.DateTime eD = new org.joda.time.DateTime(end_date);
+        Duration durationInMinutes = Duration.standardMinutes(duration);
+        Interval first_interval = Scheduler.GetOneTimeSlotBetweenDatesOfLength(getFreeBusy(start_date, end_date), durationInMinutes, sD, eD);
+        Date event_start_date = first_interval.getStart().toDate();
+        return new DateTime(event_start_date);
+    }
+
+    public Vector<Interval> getFreeBusy(Date startDate, Date endDate) throws IOException {
+        FreeBusyRequest req = new FreeBusyRequest();
+        req.setItems(getFreeBusyItems(getAllCalendarId()));
+        TimeZone tz = TimeZone.getTimeZone(timeZone);
+        DateTime startTime = new DateTime(startDate, tz);
+        DateTime endTime = new DateTime(endDate, tz);
```

Comment:
could this be in a bit more consistent naming scheme? tz, startTime, endTime
I mean you even use 'timeZone' as a variable, why not stick to that?